### PR TITLE
Dev refine class

### PR DESCRIPTION
**Pull request by alghwoorithm**

### CODE REVIEW

File: devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/AbstractDevice.java [hunk 1]
```diff
@@ -24,7 +31,9 @@
     private String nodeId;
     private String propertiesString;
     protected HashMap<String, String> propertiesMap;
+    protected HashMap<String, Object> capabilityMap;
     private DeviceType m_deviceType;
+    protected Command command;
```
Comment (bipoegg) [inline devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/AbstractDevice.java]: The same type needs to be grouped. protected, private

File: devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/AbstractDevice.java [hunk 2]
```diff
@@ -88,4 +90,101 @@ public void setCapability(String capabilityName, String capabilityValue) {
...
+    {
+        if (isCapable(Capability.KEY_SWITCH))
+        {
+            return (CapSwitch) getCapability(Capability.KEY_SWITCH);
+        }
+        else
+        {
+            return null;
+        }
+    }
```
Comment (bipoegg) [inline devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/AbstractDevice.java]: Do we need new line?

File: devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/DeviceCapabilities/CapSwitch.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        on = false;
+    }
+
+    public CapSwitch(boolean inOn) {
+        on = inOn;
+    }
+
+    public synchronized void setOn(boolean inOn)
+    {
+        on = inOn;
```
Comment (bipoegg) [inline devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/DeviceCapabilities/CapSwitch.java]: isOn?

File: devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/DeviceCapabilities/CapDoorLock.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    private boolean locked;
+
+    public CapDoorLock()
+    {
+        locked = false;
+    }
+
+    public synchronized void setLock(boolean inLocked)
+    {
+        locked = inLocked;
```
Comment (bipoegg) [inline devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/DeviceCapabilities/CapDoorLock.java]: isLocked?

File: devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/AbstractDevice.java [hunk 5]
```diff
@@ -56,21 +64,15 @@ public String getPropertiesString() {
         return propertiesMap;
     }
 
-    public boolean getCapabilityByName(String capabilityName) {
+    public HashMap<String, Object> getCapabilityMap() { return capabilityMap; }
```
Comment (bipoegg) [inline devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/AbstractDevice.java]: Should this function need to be public?

File: devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/zwave/BaseZwaveDevice.java [hunk 6]
```diff
@@ -148,44 +220,76 @@ public boolean isActive() {
...
+            case MAGNET: {
+                if (isCapable(ZwaveCapability.KEY_ACCESS_CONTROL))
+                {
+                    CapAlarm capACAlarm = getCapAccessControl();
+                    if (capACAlarm.isOn() == CapAlarm.ALARM_LEVEL_ON)
+                    {
+                        bIsActive = true;
+                    }
+                }
+            }
```
Comment (bipoegg) [inline devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/zwave/BaseZwaveDevice.java]: If getXXX() has handle isCapable, do we need to check it again?

File: devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/AbstractDevice.java (hunk 1 above)
Comment (alghwoorithm) [inline devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/AbstractDevice.java]: done

File: devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/DeviceCapabilities/CapSwitch.java (hunk 3 above)
Comment (alghwoorithm) [inline devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/DeviceCapabilities/CapSwitch.java]: Replaced by inputOn

File: devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/DeviceCapabilities/CapDoorLock.java (hunk 4 above)
Comment (alghwoorithm) [inline devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/DeviceCapabilities/CapDoorLock.java]: Replaced by inputLocked

File: devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/AbstractDevice.java (hunk 5 above)
Comment (alghwoorithm) [inline devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/AbstractDevice.java]: Device manager will use it

File: devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/zwave/BaseZwaveDevice.java (hunk 6 above)
Comment (alghwoorithm) [inline devicemanagersdk/src/main/java/cc/centralink/devicemanager/sdk/zwave/BaseZwaveDevice.java]: We don't, but I  guess check null will make more crash issue by developer?

**Pull request merged**